05_syntax_lab/02.py

```diff
@@ -1,6 +1,23 @@
-"""
-Write a Python program that randomizes 7 numbers
-and prints their sum total.
-If the sum is divisable by 7, also print the word "Boom"
-"""
-
+"""
+Syntax excersize #2
+"""
+sum = 0
+number = 0
+Boom = ""
+rand = 0
+
+from random import randint
+
+while number < 7:
+   rand = randint(1, 100)
+   print rand
+   sum  += rand
+   print sum
+   number += 1
+
+if sum % 7 == 0:
+    Boom = "Boom"
+
+print "The sum of 7 random numbers between 1 and 100 is: {} {}".format(sum, Boom)
+

```

Comment:
Hi Yinon,

Fixed program attached

Moshe Kenan
Software Engineer
Desk: +972 962 4813
Mobile: +972 54 778 4813
Verint Cyber Intelligence
www.verint.com<http://www.verint.com/>

From: ynonp [mailto:notifications@github.com]
Sent: Tuesday, July 05, 2016 8:54 PM
To: ynonp/python-examples-verint-2016-07
Cc: Kenan, Moshe; Author
Subject: Re: [ynonp/python-examples-verint-2016-07] ToCode Solution: תרגול תחביר (#24)


In 05_syntax_lab/02.py<https://github.com/ynonp/python-examples-verint-2016-07/pull/24#discussion_r69607949>:

> +rand = 0

> +

> +from random import randint

> +

> +while number < 7:

> +   rand = randint(1, 100)

> +   print rand

> +   sum  += rand

> +   print sum

> +   number += 1

> +

> +if sum % 7 == 0:

> +    Boom = "Boom"

> +

> +print "The sum of 7 random numbers between 1 and 100 is: {} {}".format(sum, Boom)

> +

I can see why you used the Boom variable, but I think it turned out not that great. I think we can have a more readable version by using two variables for the two text strings and using two separate print statements, something like:

if sum % 7 == 0:

  print BOOM.format(...)

else:

  print NOBOOM.format(...)

Try to define the variables BOOM and NOBOOM without the duplication and see if you like it better

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ynonp/python-examples-verint-2016-07/pull/24/files/c96a9094acb103cc821d4254e099039ecfb945d2#r69607949>, or mute the thread<https://github.com/notifications/unsubscribe/ATXS8uwvQoDQq9bkS_zhfoeb6pGRzROlks5qSpolgaJpZM4JFVu1>.

-- 
This electronic message may contain proprietary and confidential 
information of Verint Systems Inc., its affiliates and/or subsidiaries.
The information is intended to be for the use of the individual(s) or
entity(ies) named above. If you are not the intended recipient (or 
authorized to receive this e-mail for the intended recipient), you may not 
use, copy, disclose or distribute to anyone this message or any information 
contained in this message. If you have received this electronic message in 
error, please notify us by replying to this e-mail.